Source/UI/NewUI/PaymentForm/View/ExpiryDateView.swift

```diff
@@ -128,10 +131,7 @@ public final class ExpiryDateView: UIView {
         }
 
       case 4:
-        updateExpiryDate(to: (textField.text ?? "") + replacementText)
-        dateInputView.textFieldContainer.layer.borderColor = style?.textfield.focusBorderColor.cgColor
-        return false
-
+        return validateInputChanges(Of: textField.text ?? "", newInput: replacementText)
```

Comment:
I am still unsure why you didn't like merging the text here and you insisted on passing 2 components for the validation, but won't hold that against you 🤷‍♂️ 

```suggestion
        return validateInputChanges(of: textField.text ?? "", newInput: replacementText)
```

Reply:
Just to clarify what I mean by previous message.

This would actually require separate refactoring on the method, which you may not be inclined to do anymore :) 

```suggestion
        let input = (textField.text ?? "") + replacementText
        return validateInputChanges(for: input)
```

Reply:
some cases, the input is not valid like past year last digit. so only the current text field text should be added